### PR TITLE
Bump to polkadot-v0.9.24

### PR DESCRIPTION
**Pull request by seunlanlege**

### CODE REVIEW

**Comment by benluelo** [comment]
Perhaps out of the scope of this PR, but why don't we just depend directly on our fork in the individual crates, instead of depending on paritytech/substrate and then patching to our fork?

**Review by benluelo — APPROVED** [review]

File: frame/composable-traits/src/nft/mod.rs [hunk 1]
```diff
@@ -68,17 +68,17 @@ pub trait ReferenceNft<AccountId>:
 	// `who` is owner of NFT in case of reference NFT is reported burn (so owner is lost), it is
 	// auctioned
 	fn reference_mint_into<NFTProvider, NFT>(
-		_class: &Self::ClassId,
-		_instance: &Self::InstanceId,
+		_class: &Self::CollectionId,
```
Comment (dzmitry-lahoda) [inline frame/composable-traits/src/nft/mod.rs:71]: i ok with rename, just hope it is no conflict with any other stuff

Comment (hussein-aitlahcen) [reply frame/composable-traits/src/nft/mod.rs:71]: The rename is comming from the polkadot upgrade, not from us

**Review by hussein-aitlahcen — APPROVED** [review]

File: docker/mmr-polkadot.dockerfile [hunk 2]
```diff
@@ -12,7 +12,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends apt-utils build
 	rustup update && \
 	rustup update ${NIGHTLY_VERSION} && \
 	rustup target add wasm32-unknown-unknown --toolchain ${NIGHTLY_VERSION} && \
-	git clone -b mmr-polkadot-v0.9.22 https://github.com/composableFi/polkadot
+	git clone -b mmr-polkadot-v0.9.24 https://github.com/composableFi/polkadot
```
Comment (seunlanlege) [inline docker/mmr-polkadot.dockerfile:15]:
```suggestion
	git clone -b mmr-polkadot-v0.9.22 https://github.com/composableFi/polkadot
```

**Review by KaiserKarel — APPROVED** [review]

**Comment by seunlanlege** [comment]
simnode don't do this pls

**Review by dzmitry-lahoda — DISMISSED** [review]

**Review by benluelo — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]

**Review by KaiserKarel — APPROVED** [review]

**Review by andor0 — APPROVED** [review]

**Review by haroldsphinx — APPROVED** [review]

**Pull request merged**